packages/block-library/src/query/edit/index.js

```diff
@@ -27,7 +28,34 @@ import { DEFAULTS_POSTS_PER_PAGE } from '../constants';
 import { getFirstQueryClientIdFromBlocks } from '../utils';
 
 const TEMPLATE = [ [ 'core/post-template' ] ];
-export function QueryContent( { attributes, setAttributes } ) {
+
+function ResetQueryLoopMenuItem( { clientId } ) {
+	const selectedBlockClientId = useSelect(
+		( select ) => select( blockEditorStore ).getSelectedBlockClientId(),
+		[]
+	);
+	const { replaceInnerBlocks } = useDispatch( blockEditorStore );
+	if ( selectedBlockClientId !== clientId ) {
+		return null;
+	}
```

Comment:
Sorry, Nik, if the code is still in the works. But any reason for not using `isSelected` from blocks edit params?

Reply:
Good catch George, thanks! I had the impression that `isSelected` would return `true` on `multi-selection` as well, but it seems it doesn't.